src/components/ReportArticle.jsx

```diff
@@ -0,0 +1,120 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import { withRouter, Redirect } from 'react-router-dom';
+import propTypes from 'prop-types';
+import { toast } from 'react-toastify';
+import { reportArticle, clearErrors } from '../actions/reportArticleAction';
+
+
+class ReportArticle extends React.Component {
+  /**
+   * Creates an instance of ReportArticle class.
+   * @param {*} {props} - various props supplied to the component
+   * @memberof ReportArticle
+   */
+  constructor(props) {
+    super(props);
+    this.state = {
+      type: '',
+      comment: '',
+    };
+  }
+
+  componentDidMount() {
+    const { success } = this.props;
+    if (success) {
+      return (
+        <Redirect to="/" />
+      );
+    }
+  }
+
+  /**
+   * @param e - The input event
+   * @memberof ReportArticle
+   */
+  handleChange = e => this.setState({ [e.target.name]: e.target.value.trim() });
+
+  handleSubmit = () => {
+    const { type, comment } = this.state;
+    const { reportArticle, match: { params } } = this.props;
+    const articleId = params.id;
+    const reportData = {
+      reportType: type,
+      comment,
+      articleId,
+    };
+    reportArticle(reportData);
+  }
+
+  render() {
+    const {
+      loading, error, success, clearErrors
+    } = this.props;
+    if (error) {
+      toast.error(error);
+      clearErrors();
+    }
+    if (success) {
```

Comment:
what's the difference between this conditional and the one in the component did mount?